### PR TITLE
Double roundtrip

### PR DESCRIPTION
**Pull request by dcurrie**
I am beginning work on a pluggable atod and dtoa in benz, issue #319. The double-roundtrip branch in my repo is where I'm working, and tied to this pull request. The patch is incomplete (though it should be safe to pull). I made a new atof() function in compat.h so read_unsigned always uses atof() now. This atof() will be c-macro-ized as the "plug" in a future version.

### CODE REVIEW

**Comment by dcurrie** [comment]
Last commit has bug fix in compat.h `atof` and the refactor for `dtoa`.

**Comment by dcurrie** [comment]
Added the dtia/atod code.

I can't see why one of four test build checks is failing. I am building here with clang x64, the configuration that failed on Travis, and tests pass. Error is in ir-macro.scm !?

**Comment by dcurrie** [comment]
Ah, found the problem. Ready for review.

File: extlib/benz/string.c [hunk 1]
```diff
@@ -5,484 +5,484 @@
...
 struct pic_chunk {
-  char *str;
-  int refcnt;
-  size_t len;
-  char buf[1];
+	char *str;
+	int refcnt;
+	size_t len;
+	char buf[1];
 };
```
Comment (KeenS) [inline extlib/benz/string.c]: please use 2 spaces indent.

**Comment by KeenS** [comment]
I left a comment but basically it is a beautiful implementation. I think we can merge this patch after indentation has been fixed.
@wasabiz What's your opinion?

**Pull request merged**